include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -1070,12 +1070,44 @@ template <> class ContiguousInternalMemoryAlgorithmDataFacade<MLD> : public Algo
         InitializeInternalPointers(allocator->GetLayout(), allocator->GetMemory());
     }
 
-    const partition::MultiLevelPartitionView &GetMultiLevelPartition() const override
+    LevelID GetQueryLevel(NodeID source, NodeID target, NodeID node) const override
     {
-        return mld_partition;
+        return mld_partition.GetQueryLevel(source, target, node);
     }
 
-    const partition::CellStorageView &GetCellStorage() const override { return mld_cell_storage; }
+    LevelID GetHighestDifferentLevel(NodeID first, NodeID second) const override
+    {
+        return mld_partition.GetHighestDifferentLevel(first, second);
+    }
+
+    void ForEachSourceNodes(LevelID level, NodeID node, NodeFnT const &fn) const override
+    {
+        const auto &cell = mld_cell_storage.GetCell(level, mld_partition.GetCell(level, node));
+        auto source = cell.GetSourceNodes().begin();
+        auto durations = cell.GetInDuration(node).begin();
+
+        BOOST_ASSERT(cell.GetSourceNodes().size() == cell.GetInDuration(node).size());
```

Comment:
Add assert for GetInWeight().
Please, separate asserts with blank lines before and after.